ykrt/src/compile/jitc_yk/codegen/mod.rs

```diff
@@ -0,0 +1,218 @@
+//! The JIT's Code Generator.
+
+// FIXME: eventually delete.
+#![allow(dead_code)]
+
+use super::{jit_ir, CompilationError};
+use dynasmrt::ExecutableBuffer;
+#[cfg(any(debug_assertions, test))]
+use dynasmrt::{AssemblyOffset, DynasmApi};
+#[cfg(any(debug_assertions, test))]
```

Comment:
Why always `debug_assertions` or `test`?

Reply:
- tests use the asmprinter to match the textual representation
 - it's also useful for debugging
 - it costs to make the disassembly, so I opted to not have this functionality in release builds, but we could re-think that?

Reply:
Just to expand on this: the expensive part of making a disassembly is adding the comment lines that:

 - make it easy/possible to match only the desired part of the trace using fm tests.
 - make a trace easier for a human to read when debugging or amending tests.

E.g. a disassembly looks something like this:
```
; prologue
mov rsp, rbp
sub rsp, 64
; %0 = loadarg 0
mov ...
; %1 = load %0
mov ...
mov ...
```

Without the comment lines:
 - testing would be hard because you might match code that isn't what you want to match. comment lines serve as an anchor to an instruction of interest.
 - reading a trace is hard.

The comment lines are stored in a hashmap, and I think we don't want to be doing a load of hashing in release mode.

Reply:
Got it.

Reply:
This does raise an annoying consequence that is niggling: it means we can't reliably test release builds, and that seems backwards.

I wonder what we should do?